src/control/Control.Layers.js

```diff
@@ -165,8 +165,13 @@ export const Layers = Control.extend({
 
 	// @method collapse(): this
 	// Collapse the control container if expanded.
-	collapse() {
-		this._container.classList.remove('leaflet-control-layers-expanded');
+	collapse(ev) {
+		// On touch devices `pointerleave` is fired while clicking on a checkbox.

```

Comment:
Out of curiosity: have you tried with `pointerout`? It has a similar but *slightly* different behaviour than `pointerleave`, for this kind of situations.

Reply:
Just tested, is `pointerout` is fired same as `pointerleave`